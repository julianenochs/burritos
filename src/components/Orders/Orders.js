import React, { Component } from 'react';
import './Orders.css';
import { connect } from 'react-redux';
import { getOrders } from '../../apiCalls';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions';

export class Orders extends Component {
  constructor(props) {
    super();
    this.props = props
  }

  componentDidMount() {
    getOrders()
      .then(data => this.props.setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

  render() {
    const orderEls = this.props.orders.map(order => {
      return (
        <div className="order">
        <h3>{order.name}</h3>
        <ul key={Date.now()}className="ingredient-list">
          {order.ingredients.map(ingredient => {
            return <li key={Date.now()} >{ingredient}</li>
          })}
        </ul>
      </div>
    )
  });
    return (
      <section>
        { orderEls.length ? orderEls : <p>No orders yet!</p> }
      </section>
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);