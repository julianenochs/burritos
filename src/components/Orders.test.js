import { Orders, mapStateToProps, mapDispatchToProps, setOrders } from './Orders/Orders';
import { shallow } from 'enzyme';
import React from 'react';

describe('Orders', () => {
  let wrapper;
  let orders = [{name: 'Juju', ingredients: ['QUESOOoOoOo']}]
  beforeEach(() => {
    wrapper = shallow(<Orders orders={orders}/>)
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  describe('mapStateToProps', () => {
    it('should return state with a new order', () => {
      const mockState = {
        orders: [{ name: 'Juju', ingredients: ['QUESO0O0'] }]
      }
      const expected = {
        orders: [{ name: 'Juju', ingredients: ['QUESO0O0'] }]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected);
    });
  });

});