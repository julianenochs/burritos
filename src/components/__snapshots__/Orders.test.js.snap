// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section
  key="1573065363312"
>
  <div
    className="order"
    key="1573065363311"
  >
    <h3>
      Juju
    </h3>
    <ul
      className="ingredient-list"
      key="1573065363312"
    >
      <li
        key="1573065363312"
      >
        QUESOOoOoOo
      </li>
    </ul>
  </div>
</section>
`;
